records/filter.record.ts

```diff
@@ -0,0 +1,141 @@
+import { pool } from "../config/db";
+import { AvailableStudent, FilterEntity } from "../types";
+import { FieldPacket } from "mysql2";
+
+
+type AvailableStudentResults = [AvailableStudent[], FieldPacket[]];
+type AllRecordsStudentResults = [{totalCount:number}[],FieldPacket[]];
+
+export class FilterRecord implements FilterEntity{
+    remoteWork: boolean|string;
+    inOffice: boolean|string;
+    employmentContract: boolean|string;
+    b2b: boolean|string;
+    mandateContract: boolean|string;
+    workContract: boolean|string;
+    min: string|number;
+    max: string|number;
+    canTakeApprenticeship: boolean|null|string;
+    monthsOfCommercialExp: string|null;
+    courseCompletion: string|number;
+    courseEngagement: string|number;
+    projectDegree: string|number;
+    teamProjectDegree: string|number;
+    page: string|number;
+    rowsPerPage : string|number;
+    hrId?: string;
+
+    constructor(obj:FilterEntity) {
+        //@TODO można dodać walidacje danych
+        this.remoteWork = obj.remoteWork;
+        this.inOffice = obj.inOffice;
+        this.employmentContract = obj.employmentContract;
+        this.b2b = obj.b2b;
+        this.mandateContract = obj.mandateContract;
+        this.workContract = obj.workContract;
+        this.min = obj.min;
+        this.max = obj.max;
+        this.canTakeApprenticeship = obj.canTakeApprenticeship;
+        this.monthsOfCommercialExp = obj.monthsOfCommercialExp;
+        this.courseCompletion = obj.courseCompletion;
+        this.courseEngagement = obj.courseEngagement;
+        this.projectDegree = obj.projectDegree;
+        this.teamProjectDegree = obj.teamProjectDegree;
+        this.page = obj.page;
+        this.rowsPerPage = obj.rowsPerPage;
+        this.hrId=obj.hrId;
+    }
+
+    change(){
+        let query = "";
+        if(this.remoteWork === 'true' || this.inOffice === 'true'){
+            query +=  "`expectedTypeWork` IN (:remoteWork, :inOffice)";
+        }
+        if(this.employmentContract === true || this.b2b === true || this.mandateContract === true || this.workContract === true){
+            if(query !== ""){
+                query += " AND ";
+            }
+            query += "`expectedContractType` IN (:employmentContract, :b2b,:mandateContract,:workContract)"
+        }
+        if(this.canTakeApprenticeship !== 'null'){
+            if(query !== ""){
+                query += " AND ";
+            }
+            query += "`canTakeApprenticeship` = :canTakeApprenticeship"
+        }
+        if(query !== ""){
+            query += " AND ";
+        }
+        this.remoteWork = this.remoteWork === 'true' ? '2' : '9';
```

Comment:
Tutaj i w podobnych sytuacjach na pewno przydałoby się jakieś rzutowanie. Bo dlaczego `true` wisi jako string (rozumiem, że tak przychodzi z FE, ale to można rzutować wcześniej). Poza tym są tu używane magiczne liczby (`2`, `9`). Takie coś jest niedopuszczalne w kodzie. To się powinno opierać na enumach.